lib/propolis/src/accessors.rs

```diff
@@ -527,8 +558,15 @@ impl<T> Accessor<T> {
     /// # Panics
     ///
     /// If this is called on a non-root node.
-    pub fn poison(&self) -> Option<Arc<T>> {
-        self.0.poison()
+    pub fn poison(&self) -> Option<T::Root> {
```

Comment:
calling this `poison` makes me think `set_root_resource(Some(_))` after setting it to `None` should be an error. that would also be a stronger commitment that `Tree::new(None)` is really only useful for constructing a tree of nodes and adopting it into another three with a resource (which seems fine?)

i'd wondered about just naming this `remove_resource()`, but being able to say we're done with this tree and assert if that's incorrect seems much more valuable than "detach a resource and cross our fingers that we don't reuse this later"

Reply:
`remove_resource()` works for me.  I've done the related renaming as well.